util/xs_scripts/parallel_sim.sh

```
@@ -13,7 +31,32 @@ fi
 
-export arch_script=`realpath $1`
+# Setup gem5 environment (similar to common.sh)
+script_dir=$(dirname -- "$( readlink -f -- "$0"; )")
+export gem5_home=$(dirname $(dirname $script_dir))
+export GEM5_BUILD_TYPE=${GEM5_BUILD_TYPE:-opt}
+export gem5=$(realpath $gem5_home/build/RISCV/gem5.$GEM5_BUILD_TYPE)
+
+echo "Using gem5 binary: $gem5"
+
+# Detect if first parameter is a script (.sh) or config file (.py)
+export first_param=`realpath $1`
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)